### PR TITLE
feat: parser If Expression

### PR DESCRIPTION
**Pull request by JhaelArce**
## Description

Provide a small description of Pull Request.
The implementation of the parser for the if expression was carried out
## Type of Change

Put an `x` in all the boxes that apply:

- [X ] ✨ New Feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug Fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking Change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code Refactor
- [ ] ✅ Build Configuration Change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

## Pull Request Checklist

Put an `x` in all the boxes that apply:

- [X ] My commit messages are detailed.
- [ ] My code follows the code style of this project.
- [ ] No existing features have been broken without good reason.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests pass.

### CODE REVIEW

**Review by fabianromeroclaros — APPROVED** [review]

**Review by SebasthianSalpozz — APPROVED** [review]
I would have liked more tests on the if expression but Good job!! :D

**Review by teran-joseluis — APPROVED** [review]
Good job!

**Comment by teran-joseluis** [comment]
@JhaelArce Please update your branch to develop. You had files which are not required and we will overwrite them if I merge it.

File: package.yaml [hunk 1]
```diff
@@ -23,6 +23,7 @@ dependencies:
 - base >= 4.7 && < 5
 - parsec
 - hspec
+- HUnit
```
Comment (teran-joseluis) [inline package.yaml:26]: @JhaelArce Please update your branch to develop. You had files which are not required and we will overwrite them if I merge it.

**Review by teran-joseluis — CHANGES_REQUESTED** [review]

File: .vscode/tasks.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
+
```
Comment (teran-joseluis) [inline .vscode/tasks.json:1]: Please don't commit this directory please.

**Review by teran-joseluis — CHANGES_REQUESTED** [review]

**Review by teran-joseluis — APPROVED** [review]
Well done!

**Pull request merged**